qt-models/cylindermodel.cpp

```diff
@@ -350,100 +351,116 @@ bool CylindersModel::setData(const QModelIndex &index, const QVariant &value, in
 	QString vString = value.toString();
 	bool changed = vString != data(index, role).toString();
 
+	std::string newType; // If we allocate a new type string, this makes sure that it is freed at the end of the function
```

Comment:
that seems a bit subtle (or maybe that's just me).
Why not `strdup()` it and free it at the end? I think that would be more easily readable for me.
But maybe I'm missing other benefits of doing it this way?

Reply:
> Why not strdup() it and free it at the end?

That's functionally equivalent, however...

> But maybe I'm missing other benefits of doing it this way?

Yes - the crucial advantage is that it doesn't leak when someone adds an early return from the function. No need for `goto out` or additional `free()`, etc. The compiler is much better at such things.